### ISSUE TITLE
LSTM forget bias must be initialized properly

### ISSUE THREAD

**Issue by hengfun** [pytorch/pytorch#20102]
## 🚀 Feature
LSTM forget bias must be initialized to 1 or 2 for better training.

## Motivation

Please see:
https://pdfs.semanticscholar.org/1154/0131eae85b2e11d53df7f1360eeb6476e7f4.pdf
http://proceedings.mlr.press/v37/jozefowicz15.pdf

"This problem is addressed by simply initializing the forget
gates bf to a large value such as 1 or 2. By doing so, the
forget gate will be initialized to a value that is close to 1,
enabling gradient flow. This idea was present in Gers et al.
(2000), but we reemphasize it since we found many practitioners to not be familiar with it."

**Comment by Kaixhin** [comment]
Noted in https://github.com/pytorch/pytorch/issues/9572. The issue is currently closed, but hopefully there will be more customisable RNNs with this as an option for LSTMs.

**Comment by ezyang** [comment]
Related to #18182

**Comment by ezyang** [comment]
Yes, we'd accept this change, but in the context of the BC knob proposed and accepted in #18182 so you won't be able to change this until the BC knob is created.

**Comment by sharvil** [comment]
If you're still looking for a configurable LSTM forget bias gate, [Haste](https://github.com/lmnt-com/haste) supports it out-of-the-box and gets comparable (if not better) speeds than other fused CUDA implementations.

**Comment by loveis98** [comment]
Hi! Had this feature been released? From which pytorch version? @hengfun @ezyang @Kaixhin

**Comment by Kaixhin** [comment]
@loveis98 I believe not, the [docs](https://pytorch.org/docs/stable/generated/torch.nn.LSTMCell.html#torch.nn.LSTMCell) indicate all weights and biases are initialised with the same formula (scaled uniform noise).
